### PR TITLE
[improve][io] Make connectors load sensitive fields from secrets

### LINKED ISSUE #21674: [feat] Support load sensitive fields from secret for connectors
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Now, although some connectors' config has annotated sensitive fields with `sensitive = true`, but they still read those fields from raw configs, so It's better to make them use the `IOConfileUtils.loadWithSecrets` to load config

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiangpengcheng**
Fixes #21674

Main Issue: #xyz

PIP: #xyz 

### Motivation

Some connectors are using config to pass sensitive fields , like `token`, `password`, which is not secure

### Modifications

use the `IOConfileUtils.loadWithSecrets` to load config for connectors

### Verifying this change

- [x] Make sure that the change passes the CI checks.

- [x] This change added tests and can be verified as follows:

  - *Added unittests for connectors load sensitive fields from secrets*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/jiangpengcheng/pulsar/pull/18

### CODE REVIEW

**Review by shibd — COMMENTED** [review]
Overall LGTM. left some small comments.

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaSourceConfig.java [hunk 1]
```diff
@@ -152,8 +153,13 @@ public static KafkaSourceConfig load(String yamlFile) throws IOException {
...
 
-    public static KafkaSourceConfig load(Map<String, Object> map) throws IOException {
+    public static KafkaSourceConfig load(Map<String, Object> map, SourceContext sourceContext) throws IOException {
         ObjectMapper mapper = new ObjectMapper();
+        try {
+            map.put("sslTruststorePassword", sourceContext.getSecret("sslTruststorePassword"));
+        } catch (Exception e) {
+            // ignore
+        }
         mapper.enable(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT);
```
[the remark this answers is not in the document]
Comment (jiangpengcheng) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaSourceConfig.java:163]: not sure whether this option has effect on other connectors configs, so I only keep it here for KafkaSource only

File: pulsar-io/influxdb/src/test/java/org/apache/pulsar/io/influxdb/v1/InfluxDBSinkConfigTest.java [hunk 2]
```diff
@@ -85,12 +123,13 @@ public final void validValidateTest() throws IOException {
...
-        InfluxDBSinkConfig config = InfluxDBSinkConfig.load(map);
+        SinkContext sinkContext = Mockito.mock(SinkContext.class);
+        InfluxDBSinkConfig config = InfluxDBSinkConfig.load(map, sinkContext);
         config.validate();
     }
 
-    @Test(expectedExceptions = NullPointerException.class,
-        expectedExceptionsMessageRegExp = "influxdbUrl property not set.")
+    @Test(expectedExceptions = IllegalArgumentException.class,
+        expectedExceptionsMessageRegExp = "influxdbUrl cannot be null")
```
[the remark this answers is not in the document]
Comment (jiangpengcheng) [reply pulsar-io/influxdb/src/test/java/org/apache/pulsar/io/influxdb/v1/InfluxDBSinkConfigTest.java:132]: I think leaving it to act as a double-check would be better?

File: pulsar-io/common/src/main/java/org/apache/pulsar/io/common/IOConfigUtils.java [hunk 3]
```diff
@@ -77,13 +77,14 @@ private static <T> T loadWithSecrets(Map<String, Object> map, Class<T> clazz,
                         }
                     }
                     configs.computeIfAbsent(field.getName(), key -> {
-                        if (fieldDoc.required()) {
```
[the remark this answers is not in the document]
Comment (jiangpengcheng) [reply pulsar-io/common/src/main/java/org/apache/pulsar/io/common/IOConfigUtils.java:80]: added

File: pulsar-io/influxdb/src/test/java/org/apache/pulsar/io/influxdb/v1/InfluxDBSinkConfigTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jiangpengcheng) [reply pulsar-io/influxdb/src/test/java/org/apache/pulsar/io/influxdb/v1/InfluxDBSinkConfigTest.java:132]: ok, I'll remove them

**Review by shibd — APPROVED** [review]

**Comment by lhotari** [comment]
I think that this is a very useful improvement, but does this introduce a breaking change for existing users of the connectors?

**Pull request merged**